business/src/main/java/com/picsart/business/feature/posts/data/datasource/NetworkGetPostsDataSource.kt

```diff
@@ -13,5 +23,13 @@ internal class NetworkGetPostsDataSource(
     private val endpoint: String,
     private val httpClient: HttpClient
 ) : GetListDataSource<PostDTO> {
-    override suspend fun invoke(): Either<Failure, List<PostDTO>> = TODO()
+    override suspend fun invoke(): Either<Failure, List<PostDTO>> =
+        Either.catch { httpClient.get("$endpoint/posts").body<List<PostDTO>>() }
```

Comment:
I recommend specifying the it in both mapLeft and flatMap so it's more clear for more junior people